libs/parallelism/algorithms/include/hpx/parallel/algorithms/partition.hpp

```diff
@@ -136,21 +585,22 @@ namespace hpx { namespace parallel { inline namespace v1 {
             {
             }
 
-            template <typename ExPolicy, typename BidirIter, typename F,
-                typename Proj>
+            template <typename ExPolicy, typename BidirIter, typename Sent,
+                typename F, typename Proj>
             static BidirIter sequential(
-                ExPolicy&&, BidirIter first, BidirIter last, F&& f, Proj&& proj)
+                ExPolicy&&, BidirIter first, Sent last, F&& f, Proj&& proj)
             {
-                return std::stable_partition(first, last,
+                auto last_iter = detail::advance_to_sentinel(first, last);
+                return std::stable_partition(first, last_iter,
```

Comment:
This is a simple, but sub-optimal solution. For non-random-access iterators it requires to go over the sequence twice.

Reply:
Since the stable_partition implementation isn't trivial, would it be okay if I add it in a seperate PR? Or would you prefer I add it in this PR itself?

Reply:
Sure, np.